pkg/api/scylla/v1alpha1/types_monitoring.go

```diff
@@ -162,6 +162,16 @@ type ScyllaDBMonitoringSpec struct {
 
 	// components hold additional config for the monitoring components in use.
 	Components *Components `json:"components"`
+
+	// type determines what platform service model the monitoring is deployed for. Defaults to Platform. Valid options
+	// are Platform and SaaS.
+	// "Platform" means the monitoring is deployed in the ScyllaDB cluster owner's platform. A full monitoring setup
+	// designed for cluster administrators is deployed.
+	// "SaaS" means the monitoring is deployed for a SaaS offering. A simplified monitoring setup is deployed.
+	// +kubebuilder:default:="Platform"
```

Comment:
sounds like breaking change

Reply:
Would you rather have no default then and make it required or use "SaaS" as default?

Reply:
@tnozicka ping

Reply:
in v1alpha1 it should behave the same. with "alpha" you can break when introducing v1alpha2

it needs to be set in a way that it keeps behaving the same way (if you have deployed it already it shouldn't change your dashboards type when updating the operator+crd), which would be default=SaaS, at least for v1alpha1

Reply:
Right. I changed the default to `SaaS`, but explicitly set `Platform` in out examples. I believe it would make sense to change the default to `Platform` with the next API version though.

Reply:
I feel the same, my point was purely about API compatibility guarantees